src/Services/CheckoutService.php

```diff
@@ -1,251 +0,0 @@
-<?php
-
-declare(strict_types=1);
-
-namespace Tipoff\Checkout\Services;
-
-use Illuminate\Database\Eloquent\Model;
-use Tipoff\Checkout\Models\Cart;
-
-class CheckoutService
-{
-    /* Tax & fee application methods */
-    const APPLICATION_ORDER = 'order';
-    const APPLICATION_EACH = 'each';
-    const APPLICATION_PRODUCT = 'product';
-    const APPLICATION_BOOKING = 'booking';
-    const APPLICATION_PARTICIPANT = 'participant';
-
-    /**
-     * Check if discount/voucher code is valid.
-     *
-     * @param string $code
-     * @return mixed
-     * @psalm-suppress UndefinedMethod
-     */
-    public function findDeductionByCode($code)
-    {
-        if (strlen($code) < 4 && strlen($code) > 16) {
-            return false;
-        }
-
-        if ($voucher = app('voucher')::validAt()->where('code', $code)->first()) {
-            return $voucher;
-        }
-
-        if ($discount = app('discount')::available()->where('code', $code)->first()) {
-            return $discount;
-        }
-
-        return false;
-    }
-
-    /**
-     * Apply discount or voucher code to cart.
-     *
-     * @param string $code
-     * @return Cart
-     */
-    public function applyCodeToCart($code, $cart)
-    {
-        $deduction = $this->findDeductionByCode($code);
-
-        if (empty($deduction)) {
-            throw new \Exception("Code {$code} is invalid.");
-        }
-
-        $cart->applyDeduction($deduction);
-
-        return $cart;
-    }
-
-    /**
-     * Apply deduction to cart.
-     *
-     * @param Model $deduction
-     * @param Cart $cart
-     * @return Cart
-     */
-    public function applyDeductionToCart($deduction, $cart)
-    {
-        // TODO - temporary ugliness, will be moved into discounts/vouchers services
-        if (is_a($deduction, app('discount'))) {
-            $cart->applyDiscount($deduction);
-        }
-
-        if (is_a($deduction, app('voucher'))) {
-            $cart->applyVoucher($deduction);
-        }
-
-        return $cart;
-    }
-
-    /**
-     * Apply discount to cart.
-     *
-     * @param Model $discount
-     * @param Cart $cart
-     * @return Cart
-     */
-    public function applyDiscountToCart($discount, $cart)
-    {
-        if (! in_array($discount->applies_to, [self::APPLICATION_ORDER, self::APPLICATION_PARTICIPANT])) {
-            return $cart;
-        }
-
-        $cart->discounts()->syncWithoutDetaching([$discount->id]);
-        $cart->total_deductions = $this->calculateCartDeductions($cart);
-        $cart->save();
-
-        return $cart;
-    }
-
-    /**
-     * Issue partial redemption voucher when total discounts are higher than total amount.
-     *
-     * @param  Cart $cart
-     * @return Model
-     */
-    public function issueCartPartialRedemptionVoucher($cart)
```

Comment:
Gets of implementation moved to Voucher package  - `VoucherInterface::issuePartialRedemptionVoucher(int $locationId, int $amount, int $userId)` - note the interface is explicit in the parameters it wants.